gui/src/components/LoadingBar.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import LinearProgress from '@material-ui/core/LinearProgress'
+import { withStyles } from '@material-ui/core/styles'
+
+const styles = theme => ({
+  root: {
+    flexGrow: 1,
+    height: 3,
+    overflow: 'hidden'
+  },
+  colorPrimary: {
+    backgroundColor: theme.palette.background.default
+  },
+  barColorPrimary: {
+    backgroundColor: theme.palette.primary.main
+  }
+})
+
+class LoadingBar extends Component {
+  state = {
+    completed: 0
+  };
+
+  componentDidMount () {
+    this.timer = setInterval(this.progress, 500)
+  }
+
+  componentWillUnmount () {
+    clearInterval(this.timer)
+  }
+
+  progress = () => {
+    const { completed } = this.state
+    if (completed === 100) {
+      this.setState({ completed: 0 })
+    } else {
+      const diff = Math.random() * 10
+      this.setState({ completed: Math.min(completed + diff, 100) })
```

Comment:
Hi there @rupurt, can you elaborate on this method a bit. I see that you took it from the `material-ui` example.

https://material-ui.com/demos/progress/#linear-determinate

Is it because we prefer this random linear aesthetic to the busy `indeterminate` one? And yet, we still set the `variant` to `indeterminate`.

Reply:
@dimroc updated with math removed.